### PR TITLE
[Excel] (Range) Add precedents API to existing article

### PR DESCRIPTION
**Pull request by alison-mk**

### CODE REVIEW

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md [hunk 1]
```diff
@@ -12,40 +12,57 @@ Excel formulas often refer to other cells. These cross-cell references are known
...
 
-## Get the direct precedents of a formula
+## Get the precedents of a formula
 
-Locate a formula's direct precedent cells with [Range.getDirectPrecedents](/javascript/api/excel/excel.range#getDirectPrecedents__). `Range.getDirectPrecedents` returns a `WorkbookRangeAreas` object. This object contains the addresses of all the direct precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one formula precedent. For more information on working with the `RangeAreas` object, see [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
+Locate all of a formula's precedent cells with [Range.getPrecedents](/javascript/api/excel/excel.range#getPrecedents__). `Range.getPrecedents` returns a `WorkbookRangeAreas` object. This object contains the addresses of all the precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one formula precedent. For more information on working with the `RangeAreas` object, see [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
```
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-precedents-dependents.md:19]: Two minor suggestions for brevity:
```suggestion
Locate a formula's precedent cells with [Range.getPrecedents](/javascript/api/excel/excel.range#getPrecedents__). `Range.getPrecedents` returns a `WorkbookRangeAreas` object. This object contains the addresses of all the precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one formula precedent. To learn more about the `RangeAreas` object, see [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
```

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md [hunk 2]
```diff
@@ -12,40 +12,57 @@ Excel formulas often refer to other cells. These cross-cell references are known
...
+To locate only the direct precedent cells of a formula, use [Range.getDirectPrecedents](/javascript/api/excel/excel.range#getDirectPrecedents__). `Range.getDirectPrecedents` works like `Range.getPrecedents` and returns a `WorkbookRangeAreas` object containing the addresses of direct precedents.
+
+The following screenshot shows the result of selecting the **Trace Precedents** button in the Excel UI. This button draws an arrow from precedent cells to the selected cell. The selected cell, **E3**, contains the formula "=C3 * D3", so both **C3** and **D3** are precedent cells. Unlike the Excel UI button, the `getPrecedents` and `getDirectPrecedents` methods don't draw arrows.
 
 ![Arrow tracing precedent cells in the Excel UI.](../images/excel-ranges-trace-precedents.png)
 
 > [!IMPORTANT]
-> The `getDirectPrecedents` method can't retrieve precedent cells across workbooks.
+> The `getPrecedents` and `getDirectPrecedents` methods can't retrieve precedent cells across workbooks.
```
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-precedents-dependents.md:28]: Something about "can't" makes it sound like that was the desired behavior.
```suggestion
> The `getPrecedents` and `getDirectPrecedents` methods do not retrieve precedent cells across workbooks.
```

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md [hunk 3]
```diff
@@ -12,40 +12,57 @@ Excel formulas often refer to other cells. These cross-cell references are known
...
 
-The following code sample gets the direct precedents for the active range and then changes the background color of those precedent cells to yellow.
+The following code sample shows how to use the `Range.getPrecedents` and `Range.getDirectPrecedents` methods to get the precedents for the active range and then change the background color of those precedent cells. The sample changes the background color of the direct precedent cells to yellow and the background color of the other precedent cells to orange.
```
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-precedents-dependents.md:30]: The first sentence seemed long. Consider splitting it up along these lines.
```suggestion
The following code sample shows how to work with the `Range.getPrecedents` and `Range.getDirectPrecedents` methods. The sample gets the precedents for the active range and then changes the background color of those precedent cells. The background color of the direct precedent cells is set to yellow and the background color of the other precedent cells is set to orange.
```

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md [hunk 4]
```diff
@@ -12,40 +12,57 @@ Excel formulas often refer to other cells. These cross-cell references are known
 
 A cell may have a precedent cell, and that precedent cell may have its own precedent cells. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship. A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship. Cells that refer to other cells in a workbook, but whose relationship is not a parent-child relationship, are not direct dependents or direct precedents.
```
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-precedents-dependents.md:13]: Consider removing or changing the last sentence of this paragraph. Now that the article has expanded beyond direct dependents and precedents, that sentence makes it seem like things outside of the parent-child relationship aren't relevant here, and that's not true any more.

**Review by AlexJerabek — CHANGES_REQUESTED** [review]
A couple minor suggestions and one bigger question about the text framing the article.

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md (hunk 4 above)
Comment (alison-mk) [reply docs/excel/excel-add-ins-ranges-precedents-dependents.md:13]: I'm glad you noticed this. I tried multiple variations of this sentence and didn't find a good fit. I think the sentence is valuable in that it reiterates what a direct dependent or precedent is, which is an abstract concept. What do you think of this version? 
> Cells that refer to other cells in a workbook, but whose relationship is not a parent-child relationship, are dependents or precedents, but they are not direct dependents or direct precedents.

Comment (AlexJerabek) [reply docs/excel/excel-add-ins-ranges-precedents-dependents.md:13]: Maybe the problem is we're not defining the more "complex" relationship first. How about something like this?
```suggestion
A precedent cell may have its own precedent cells. Every precedent cell in this chain of precedents is still a precedent of the original cell. The same relationship exists for dependents: any cell affected by another cell is a dependent of that other cell. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship.  A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship.
```

File: docs/excel/excel-add-ins-ranges-precedents-dependents.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ ms.localizationpriority: medium
...
 
-A cell may have a precedent cell, and that precedent cell may have its own precedent cells. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship. A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship. Cells that refer to other cells in a workbook, but whose relationship is not a parent-child relationship, are not direct dependents or direct precedents.
+A precedent cell may have its own precedent cells. Every precedent cell in this chain of precedents is still a precedent of the original cell. The same relationship exists for dependents. Any cell affected by another cell is a dependent of that cell. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship.  A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship.
```
Comment (AlexJerabek) [inline docs/excel/excel-add-ins-ranges-precedents-dependents.md:13]: Extra space.
```suggestion
A precedent cell may have its own precedent cells. Every precedent cell in this chain of precedents is still a precedent of the original cell. The same relationship exists for dependents. Any cell affected by another cell is a dependent of that cell. A "direct precedent" is the first preceding group of cells in this sequence, similar to the concept of parents in a parent-child relationship. A "direct dependent" is the first dependent group of cells in a sequence, similar to children in a parent-child relationship.
```

**Review by AlexJerabek — APPROVED** [review]
Looks good! Thanks for fixing that paragraph. Just one minor typo, but this should be good to check in.

**Pull request merged**